include/deal.II/fe/mapping_fe_field.h

```diff
@@ -32,81 +32,94 @@ DEAL_II_NAMESPACE_OPEN
- * of absolute positions instead of a vector of displacements.
- * In particular we think of a collections of a FE_Q or
- * Bezier finite element (FE_Bernstein) repeated a number of times equal to the space
- * dimension. The idea is to construct the mapping using a vector of control
- * points, a DoFHandler associated to the geometry of the problem and a
- * ComponentMask that tells us which components to use for the mapping.
- * This mapping will grab from the DoFHandler the finite element, or better
- * the collection of finite elements, to compute the mapping shape functions.
- * So we will have two different Finite Element and DoFHandler, one for the
- * solution field and one to describe the geometry of the problem. Historically
- * in the deal.II library there was not such a distinction. The differences
- * between this mapping and the MappingQ class are quite important.
- * The MappingFE, being a generalization, requires a higher level of abstraction.
- * This is the reason why it takes a DoFHandler and a vector of control points
- * that are the coefficients of the shape function (so in general it is a vector
- * of coefficient).
+ * The MappingFEField is a generalization of the MappingQEulerian
+ * class, for arbitrary vector finite elements. The two main
+ * differences are that this class uses a vector of absolute positions
+ * instead of a vector of displacements, and it allows for arbitrary
+ * FiniteElement types, instead of only FE_Q.
  *
+ * This class effectively decouples the topology from the geometry, by
+ * relegating all geometrical information to some components of a
+ * FiniteElement vector field. The components that are used for the
+ * geometry can be arbitrarily selected at construction time.
+ *
+ * The idea is to consider the Triangulation as a parameter
+ * configuration space, on which we  construct an arbitrary
+ * geometrical mapping, using the instruments of the deal.II library:
+ * a vector of degrees of freedom, a DoFHandler associated to the
+ * geometry of the problem and a ComponentMask that tells us which
+ * components of the FiniteElement to use for the mapping.
+ *
+ * Typically, the DoFHandler operates on a finite element that is
+ * constructed as a system element (FESystem()) from continuous FE_Q()
+ * (for iso-parametric discretizations) or FE_Bernstein() (for
+ * iso-geometric discretizations) objects. An example is shown below:
  *
- * Typically, the DoFHandler operates on a finite element that is constructed
- * as a system element (FESystem) from continuous FE_Q() objects. An example
- * is shown below:
  * @code
  *    const FE_Q<dim,spacedim> feq(1);
  *    const FESystem<dim,spacedim> fesystem(feq, spacedim);
  *    DoFHandler<dim,spacedim> dhq(triangulation);
  *    dhq.distribute_dofs(fesystem);
  *    Vector<double> eulerq(dhq.n_dofs());
+ *    // Fills the euler vector with information from the Triangulation
+ *    VectorTools::get_position_vector(dhq, eulerq);
```

Comment:
I see, and also that the MappingQEulerian has the same default. I think -- just for clarity -- it would be useful if the example in the documentation used a component mask either for both or none of the two calls.